src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -0,0 +1,1292 @@
+            {
+                this.jsonWriter.WriteStringValue(containedEntitySet.Path.Path);
+            }
+            else
+            {
+                this.jsonWriter.WriteStringValue(binding.Target.Name);
+            }
+        }
+
+        internal override void WriteEndElement()
+        {
+            this.jsonWriter.WriteEndObject();
+        }
+
+        internal override void WriteArrayEndElement()
+        {
+            this.jsonWriter.WriteEndArray();
+        }
+
+        internal override void WriteOperationImportAttributes(IEdmOperationImport operationImport, string operationAttributeName)
+        {
+            if (operationImport.EntitySet != null)
+            {
+                var pathExpression = operationImport.EntitySet as IEdmPathExpression;
+                if (pathExpression != null)
+                {
+                    this.jsonWriter.WriteRequiredProperty("$EntitySet", pathExpression.PathSegments, PathAsXml);
+                }
+                else
+                {
+                    throw new InvalidOperationException(Strings.EdmModel_Validator_Semantic_OperationImportEntitySetExpressionIsInvalid(operationImport.Name));
+                }
+            }
+        }
+
+        private string SerializationName(IEdmSchemaElement element)
+        {
+            if (this.NamespaceAliasMappings != null)
+            {
+                string alias;
+                if (this.NamespaceAliasMappings.TryGetValue(element.Namespace, out alias))
+                {
+                    return alias + "." + element.Name;
+                }
+            }
+
+            return element.FullName();
+        }
+
+        private string TypeDefinitionAsJson(IEdmSchemaType type)
+        {
+            return this.SerializationName(type);
+        }
+
+        private string AnnotationToString(IEdmVocabularyAnnotation annotation)
+        {
+            StringBuilder sb = new StringBuilder(CsdlConstants.Prefix_At);
+            sb.Append(SerializationName(annotation.Term));
+
+            if (annotation.Qualifier != null)
+            {
+                sb.Append(CsdlConstants.Prefix_Hash).Append(annotation.Qualifier);
+            }
+
+            if (this.isInEnumTypeWriting)
+            {
+                IEdmEnumMember enumMember = annotation.Target as IEdmEnumMember;
+                if (enumMember != null)
+                {
+                    return enumMember.Name + sb.ToString();
+                }
+            }
+
+            return sb.ToString();
+        }
+
+        protected string EnumMemberExpressionAsJson(IEnumerable<IEdmEnumMember> members)
+        {
+            if (this.settings.EnumMemberExpressionAsNumeric)
+            {
+                long result = 0;
+                foreach (var member in members)
+                {
+                    result += member.Value.Value;
+                }
+
+                return result.ToString(CultureInfo.InvariantCulture);
+            }

```

Comment:
I'd like to remove this option as it is against spec recommendation.

Reply:
removed.

---
In reply to: [443762262](https://github.com/OData/odata.net/pull/1775#discussion_r443762262) [](ancestors = 443762262)